src/index/TextBlockWriter.cpp

```diff
@@ -0,0 +1,118 @@
+#include "TextBlockWriter.h"
+
+#include "index/TextIndexReadWrite.h"
+
+template <typename WordTextVecView, typename EntityTextVecView>
+void TextBlockWriter<WordTextVecView, EntityTextVecView>::
+    calculateAndWriteTextBlocks(size_t nofWordPostingsInTextBlock) {
+  AD_CONTRACT_CHECK(
+      nofWordPostingsInTextBlock > 0,
+      "Number of word postings in text block has to be larger than zero.");
+  for (auto&& chunk :
+       ::ranges::views::chunk(wordTextVecView_, nofWordPostingsInTextBlock)) {
+    for (const auto& row : chunk) {
+      currentWordTextVecWordIndex_ = row[0].getWordVocabIndex();
+      addWordPosting(row);
+    }
+    addBlock();
+  }
+}
+
+// _____________________________________________________________________________
+template <typename WordTextVecView, typename EntityTextVecView>
+void TextBlockWriter<WordTextVecView, EntityTextVecView>::addBlock() {
+  // Add all co-occurring entities to entityPostings_
+  addEntityPostingsUpToWordIndex();
+
+  // This is possible since the wordPostings are filled in ascending WordIndex
+  // order.
+  WordVocabIndex minWordIndexOfBlock = std::get<1>(wordPostings_.front());
+  WordVocabIndex maxWordIndexOfBlock = std::get<1>(wordPostings_.back());
```

Comment:
Isn't this exactly then the argument to the `addEntityPostingsUpTo` and you can get rid of the member variable that
stores the current upper bound word index?